### PR TITLE
[ACR] `acr create/update`: Add new parameter `--allow-exports` to support export

### PR DESCRIPTION
**Pull request by adewaleo**
**Description**

[ACR] Add --allow-exports to `acr create / update`. Support for exportPolicy / export disable. Bump azure-mgmt-containerregistry to 8.1.0. 

**Testing Guide**

**History Notes**

[ACR]: `az acr create/update`: Add support for disabling export through `--allow-exports`
[ACR]: `az acr`: Bump core api-version to `2021-06-01-preview` from `2020-11-01-preview`. agent_pool, tasks and runs operations unchanged from `2019-06-01-preview`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ACR

**Comment by adewaleo** [comment]
image

image

**Review by ankurkhemani — COMMENTED** [review]
Can we please update the [HISTORY.rst](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/HISTORY.rst) to include the changes in this PR.

File: src/azure-cli/azure/cli/command_modules/acr/custom.py [hunk 1]
```diff
@@ -153,6 +157,20 @@ def _handle_network_bypass(cmd, registry, allow_trusted_services):
...
+def _handle_export_policy(cmd, registry, allow_exports):
+    if allow_exports is not None:
+        Policies, ExportPolicy, ExportPolicyStatus = cmd.get_models('Policies', 'ExportPolicy', 'ExportPolicyStatus')
+
+        if registry.policies is None:
+            registry.policies = Policies()
+
+        status = ExportPolicyStatus.DISABLED if not allow_exports else ExportPolicyStatus.ENABLED
+        try:
+            registry.policies.export_policy.status = status
```
Comment (ankurkhemani) [inline src/azure-cli/azure/cli/command_modules/acr/custom.py:169]: Can `registry.policies.export_policy` be `None`?

File: src/azure-cli/azure/cli/command_modules/acr/tests/latest/recordings/test_acr_create_with_managed_registry.yaml [hunk 2]
```diff
@@ -18,24 +18,24 @@ interactions:
       ParameterSetName:
       - -n -g -l --sku
       User-Agent:
-      - AZURECLI/2.26.0 azsdk-python-azure-mgmt-containerregistry/8.0.0 Python/3.8.3
-        (Windows-10-10.0.18362-SP0)
+      - AZURECLI/2.26.1 azsdk-python-azure-mgmt-containerregistry/8.1.0 Python/3.7.9
```
Comment (ankurkhemani) [inline src/azure-cli/azure/cli/command_modules/acr/tests/latest/recordings/test_acr_create_with_managed_registry.yaml:21]: Python version went down from `3.8.3` to `3.7.9`?

**Review by ankurkhemani — APPROVED** [review]
LGTM.

**Comment by adewaleo** [comment]
> Can we please update the [HISTORY.rst](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/HISTORY.rst) to include the changes in this PR.

@ankurkhemani I don't think we need to update history.rst. If the PR title is formatted well or description updated, that should be fine. https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#format-pr-description

**Review by zhoxing-ms — APPROVED** [review]

**Comment by adewaleo** [comment]
@zhoxing-ms Please go ahead and merge whenever you're ready. @ankurkhemani from ACR has approved.

cc: @northtyphoon @rosanch

**Comment by zhoxing-ms** [comment]
I don't have permission to merge because we still lack the approval of code owners. @Juliehzl @jsntcy please help to have a look~

**Review by Juliehzl — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Comment by ankurkhemani** [comment]
> > Can we please update the [HISTORY.rst](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/HISTORY.rst) to include the changes in this PR.
> 
> @ankurkhemani I don't think we need to update history.rst. If the PR title is formatted well or description updated, that should be fine. https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#format-pr-description

Yes this works too :)

**Pull request merged**